src/intercom/back_end_binding.py

```diff
@@ -34,90 +33,66 @@ def __init__(
         compare_service=None,
         unpacking_service=None,
         unpacking_locks=None,
-        testing=False,  # noqa: ARG002
     ):
         self.analysis_service = analysis_service
         self.compare_service = compare_service
         self.unpacking_service = unpacking_service
         self.unpacking_locks = unpacking_locks
-        self.poll_delay = config.backend.intercom_poll_delay
-
-        self.stop_condition = Value('i', 0)
-        self.process_list = []
+        self.listeners = [
+            InterComBackEndAnalysisTask(self.unpacking_service.add_task),
+            InterComBackEndReAnalyzeTask(self.unpacking_service.add_task),
+            InterComBackEndCompareTask(self.compare_service.add_task),
+            InterComBackEndRawDownloadTask(),
+            InterComBackEndFileDiffTask(),
+            InterComBackEndTarRepackTask(),
+            InterComBackEndBinarySearchTask(),
+            InterComBackEndUpdateTask(self.analysis_service.update_analysis_of_object_and_children),
+            InterComBackEndDeleteFile(
+                self.analysis_service.update_analysis_of_object_and_children,
```

Comment:
Why do I need to `update_analysis_of_object_and_children` if I want to delete a firmware?

Reply:
Good catch. This is obviously wrong. Maybe it snuck in during a rebase/merge